internal/controller/nats/client.go

```diff
@@ -0,0 +1,96 @@
+package nats
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/nats-io/nats.go"
+)
+
+type Client interface {
```

Comment:
Why dont we move this client to `pkg/nats`?

Reply:
we can, I'll move it.